apps/designer/app/designer/features/workspace/canvas-tools/text-toolbar/text-toolbar.tsx

```diff
@@ -49,16 +64,57 @@ const onClickPreventDefault: MouseEventHandler<HTMLDivElement> = (event) => {
 
 type ToolbarProps = {
   css?: CSS;
-  onValueChange: (value: Value) => void;
   rootRef: React.Ref<HTMLDivElement>;
+  textToolbar: TextToolbarValue;
+  publish: Publish;
 };
 
-const Toolbar = ({ css, onValueChange, rootRef }: ToolbarProps) => {
+const Toolbar = ({ css, rootRef, textToolbar, publish }: ToolbarProps) => {
+  const value: Value[] = [];
+  if (textToolbar.isBold) {
+    value.push("Bold");
+  }
+  if (textToolbar.isItalic) {
+    value.push("Italic");
+  }
+  if (textToolbar.isLink) {
+    value.push("Link");
+  }
   return (
     <ToggleGroup.Root
       ref={rootRef}
-      type="single"
-      onValueChange={onValueChange}
+      type="multiple"
+      value={value}
+      onValueChange={(newValues: Value[]) => {
+        // TODO refactor with per button callback
+        if (
+          textToolbar.isBold === false &&
+          newValues.includes("Bold") === true
+        ) {
+          publish({
+            type: "textToolbarFormat",
+            payload: "bold",
+          });
+        }
+        if (
+          textToolbar.isItalic === false &&
+          newValues.includes("Italic") === true
+        ) {
+          publish({
+            type: "textToolbarFormat",
+            payload: "italic",
+          });
+        }
+        if (
+          textToolbar.isLink === false &&
+          newValues.includes("Link") === true
+        ) {
+          publish({
+            type: "textToolbarFormat",
+            payload: "link",
+          });
+        }
```

Comment:
this logic could be more generic, instead of case-by-case

Reply:
I would personally prefer mui like api where each case is possible to handle separately then no need for this logic at all.

Reply:
not sure what you mean but sounds good

Reply:
something like this
```
<Group>
  <Button selected={isBold} onClick={() => {}} />
  <Button selected={isItalic} onClick={() => {}} />
  <Button selected={isLink} onClick={() => {}} />
</Group>
```

Reply:
nice